pkg/kubectl/cmd/get/get.go

```diff
@@ -733,7 +733,7 @@ func (o *GetOptions) printGeneric(r *resource.Result) error {
 	}
 
 	var obj runtime.Object
-	if !singleItemImplied || len(infos) > 1 {
+	if !singleItemImplied || len(infos) > 1 || len(infos) == 0 {
 		// we have more than one item, so coerce all items into a list.
 		// we don't want an *unstructured.Unstructured list yet, as we
 		// may be dealing with non-unstructured objects. Compose all items
```

Comment:
do we need to initialize items to `Items: []runtime.RawExtension{}` to avoid null items if they do `-o json`?

Reply:
Non initialized array will be an empty array and that's sufficient for our needs. I've added `TestEmptyResultJSON` as a security net for that case. We only print `No resources found` when using the default printers, b/c we don't have other way of saying there's nothing there. In case of JSON or other structured output you'll get an empty list.